hello/hello_cron.py

```diff
@@ -0,0 +1,64 @@
+import asyncio
+from dataclasses import dataclass
+from datetime import timedelta
+
+from temporalio import activity, workflow
+from temporalio.client import Client
+from temporalio.common import RetryPolicy
+from temporalio.worker import Worker
+
+
+@dataclass
+class ComposeGreetingInput:
+    greeting: str
+    name: str
+
+
+@activity.defn
+async def compose_greeting(input: ComposeGreetingInput) -> str:
+    return f"{input.greeting}, {input.name}!"
+
+
+@workflow.defn
+class GreetingWorkflow:
+    @workflow.run
+    async def run(self, name: str) -> None:
+        result = await workflow.execute_activity(
+            compose_greeting,
+            ComposeGreetingInput("Hello", name),
+            start_to_close_timeout=timedelta(seconds=10),
+        )
+        print(f"Result: {result}")
+
+
+async def main():
+    # Start client
+    client = await Client.connect("http://localhost:7233")
+
+    # Run a worker for the workflow
+    async with Worker(
```

Comment:
What happens if a user hits ctrl-c after the worker has started?
Does the worker gracefully shutdown?

Reply:
The worker does get gracefully shutdown because of `async with`. `Worker.__aexit__` is invoked. If a user didn't use `async with` but instead `Worker.run()`, they'd be responsible for `Worker.shutdown()` in an `except` or `finally`.

Reply:
👍